### DISCUSSION TITLE
Is there a way to execute migrations from inside of your code?

### DISCUSSION THREAD

**Discussion by timuckun** [amacneil/dbmate#233] [Feature Ideas]
I don't want to install this into the docker container and execute it.  If there is a way to do it by including a module that would be great.

**Comment by amacneil** [comment]
Yes, you can import [`pkg/dbmate`](https://github.com/amacneil/dbmate/tree/main/pkg/dbmate) to use it like a library.

All the CLI functionality is self-contained in [`main.go`](https://github.com/amacneil/dbmate/blob/main/main.go) (which you can use as an example for how to use the library).
